### PR TITLE
Signup jquery validations complete, closing user story #37

### PR DESCRIPTION
**Pull request by cluhring**
Added more HTML to signup form, jquery validations w/ css, and strong params to Company Controller.  Possibly some other stuff too.  Merged with master as of 10pm.

### CODE REVIEW

**Comment by oorion** [comment]
Looks awesome!  I really like the javascript validations.

File: Gemfile [hunk 1]
```diff
@@ -26,6 +26,7 @@ group :development, :test do
   gem 'binding_of_caller'
   gem 'pry'
   gem 'capybara'
+  gem 'capybara-screenshot', :group => :test
```
Comment (KristaANelson) [inline Gemfile:30]: Sweet!! :sparkles: So excited about this capybara-screenshot! Never heard of it before seeing it here!

File: app/assets/javascripts/jquery.validate.js [hunk 2]
```diff
@@ -0,0 +1,1365 @@
...
+			}
+			// Add aria-invalid status for screen readers
+			$( element ).attr( "aria-invalid", !result );
+
+			if ( !this.numberOfInvalids() ) {
+				// Hide error containers on last error
+				this.toHide = this.toHide.add( this.containers );
+			}
+			this.showErrors();
+			return result;
```
Comment (KristaANelson) [inline app/assets/javascripts/jquery.validate.js:438]: :+1: Nice to see this for screen readers!

File: app/assets/javascripts/jquery.validate.js [hunk 3]
```diff
@@ -0,0 +1,1365 @@
...
+		return this.bind(type, function( event ) {
+			var target = $(event.target);
+			if ( target.is(delegate) ) {
+				return handler.apply(target, arguments);
+			}
+		});
+	}
+});
+
+}));
```
Comment (KristaANelson) [inline app/assets/javascripts/jquery.validate.js:1365]: Hefty file right there! But really cool, I like that it disables the submit button until all of the errors that are addressed. I also like how easy it is to update the the error messages for the different form fields. :sparkles:

File: app/controllers/companies_controller.rb [hunk 4]
```diff
@@ -17,6 +17,13 @@ def create
...
-    params.require(:company).permit(:name, :organization, :title, :state, :city, :email )
+    params.require(:company).permit(:name,
+                                    :organization,
+                                    :title,
+                                    :state,
+                                    :city,
+                                    :email,
+                                    :hiring,
+                                    :hire_count )
   end
```
Comment (KristaANelson) [inline app/controllers/companies_controller.rb:28]: :+1:

File: app/models/company.rb [hunk 5]
```diff
@@ -2,5 +2,5 @@ class Company < ActiveRecord::Base
   validates :name, :organization, :title, :state, :city, presence: true
   validates :email, presence: true, length: { maximum: 255 },
                     format: { with: /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i, }
-
+  validates :hire_count, numericality: { only_integer: true, }
 end
```
Comment (KristaANelson) [inline app/models/company.rb:21]: Nice!! I haven't seen numerically before! I was using regex, but this is very readable and easy to use!! :sparkles:

File: app/views/application/_signup_form.html.erb [hunk 6]
```diff
@@ -2,21 +2,42 @@
...
+      <%= f.label(:hire_count, "people through TechHire") %>
+    </div>
     <%= f.submit "Create techHire account" %>
-   <% end %>
-   </div>
+    <% end %>
+    <script>
+    $("#new_company").validate();
+    </script>
 </div>
```
Comment (KristaANelson) [inline app/views/application/_signup_form.html.erb:43]: This is what triggers the validation. Pretty cool!

File: spec/features/guest_user_spec.rb [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+        select('Fort Collins', from: 'company[city]')
+        fill_in 'company[email]', with: "goog"
+        check 'company[hiring]'
+        fill_in 'company[hire_count]', with: 5
+        click_button('Create techHire account')
+      end
+      expect(current_path).to eq(companies_path)
+      expect(page.find('.errors')).to have_content("Please try again!")
+    end
+  end
```
Comment (KristaANelson) [inline spec/features/guest_user_spec.rb:65]: Happy and sad paths!! :+1:

File: app/views/application/_signup_form.html.erb (hunk 6 above)
Comment (scottalexandra) [inline app/views/application/_signup_form.html.erb:43]: Nice Work Chris!

File: Gemfile (hunk 1 above)
Comment (Lydias303) [inline Gemfile:30]: Im excited to try this out!

File: app/views/application/_signup_form.html.erb (hunk 6 above)
Comment (scottalexandra) [inline app/views/application/_signup_form.html.erb]: Nice work Chris!

File: app/assets/javascripts/jquery.validate.js (hunk 3 above)
Comment (Lydias303) [inline app/assets/javascripts/jquery.validate.js:1365]: Agreed!

File: spec/features/guest_user_spec.rb [hunk 8]
```diff
@@ -1,29 +1,66 @@
...
+        fill_in 'company[email]', with: "goog"
+        check 'company[hiring]'
+        fill_in 'company[hire_count]', with: 5
+        click_button('Create techHire account')
       end
+      expect(current_path).to eq(companies_path)
+      expect(page.find('.errors')).to have_content("Please try again!")
     end
   end
 end
```
Comment (Lydias303) [inline spec/features/guest_user_spec.rb:66]: :+1:

**Pull request merged**